dotnet-interactive.Tests/LspSerializationContractTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Runtime.CompilerServices;
+using Assent;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.App.Lsp;
+using Microsoft.DotNet.Interactive.Extensions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.App.Tests
+{
+    public class LspSerializationContractTests
+    {
+        private readonly Configuration _configuration;
+
+        public LspSerializationContractTests()
+        {
+            _configuration = new Configuration()
+                .UsingExtension("json");
+
+            _configuration = _configuration.SetInteractive(true);
+        }
+
+        private void Dissent<T>(T expected, JsonSerializer serializer, [CallerMemberName] string testName = null, [CallerFilePath] string filePath = null)
```

Comment:
After looking at it more closely, I think this makes the Assent tests confusing (since some of the `.approved` files no longer involve an "approval" process), and it makes the tests where this is used harder to understand (because the file contents are important to understanding the test, so why put them in a separate file?).

Let's just inline the JSON into the tests for these. If escaping is a pain you can use anonymous objects e.g.

```csharp
var json = new {
    contents = new {
        kind = "markdown",
        value=  "content",
        range= new {
            start = new { line = 1, character= 2 },
            end = new { line= 3, character= 4 }
    }.ToJson();
```